protobuf/src/main/java/com/fasterxml/jackson/dataformat/protobuf/schema/DescriptorReader.java

```diff
@@ -0,0 +1,200 @@
+package com.fasterxml.jackson.dataformat.protobuf.schema;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.protobuf.ProtobufMapper;
+import com.squareup.protoparser.DataType;
+import com.squareup.protoparser.DataType.ScalarType;
+import com.squareup.protoparser.EnumConstantElement;
+import com.squareup.protoparser.EnumElement;
+import com.squareup.protoparser.FieldElement;
+import com.squareup.protoparser.MessageElement;
+import com.squareup.protoparser.OptionElement;
+import com.squareup.protoparser.ProtoFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DescriptorReader
+{
+    private final String DESCRIPTOR_PROTO = "/descriptor.proto";
+    private final ProtobufMapper descriptorMapper;
+    private final ProtobufSchema descriptorFileSchema;
+
+    public DescriptorReader() throws IOException
+    {
+        // read Descriptor Proto
+        descriptorMapper = new ProtobufMapper();
+        InputStream in = getClass().getResourceAsStream(DESCRIPTOR_PROTO);
+        descriptorFileSchema = ProtobufSchemaLoader.std.load(in, "FileDescriptorSet");
+        in.close();
+    }
+
+    public Map<String, ProtoFile> readFileDescriptorSet(String descriptorFilePath) throws IOException
+    {
+        InputStream fin;
+        fin = this.getClass().getClassLoader().getResourceAsStream(descriptorFilePath);
+        if (fin == null) {
```

Comment:
I was trying to make it dual purpose -- load the descriptor from resources directory as well as external URL.    I will remove the resources.  I guess the descriptors in the resources directory is only useful for unit tests.